spec/models/waste_exemptions_engine/edit_registration_workflow_state/location_form_spec.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module WasteExemptionsEngine
+  RSpec.describe EditRegistration, type: :model do
+    describe "#workflow_state" do
+      previous_state = :edit_form
+      current_state = :location_form
+      subject(:edit_registration) { create(:edit_registration, workflow_state: current_state) }
+
+      context "when a WasteExemptionsEngine::EditRegistration's state is #{current_state}" do
+        context "when none of the should register in location conditions are true" do
+          next_state = :edit_form
+
+          before(:each) { edit_registration.location = "england" }
+
+          it "can only transition to #{next_state}" do
+            permitted_states = Helpers::WorkflowStates.permitted_states(edit_registration)
+            expect(permitted_states).to match_array([next_state])
+          end
+
+          it "changes to #{next_state} after the 'next' event" do
+            expect(edit_registration.send(:should_register_in_northern_ireland?)).to eq(false)
+            expect(edit_registration.send(:should_register_in_scotland?)).to eq(false)
+            expect(edit_registration.send(:should_register_in_wales?)).to eq(false)
+            expect(edit_registration).to transition_from(current_state).to(next_state).on_event(:next)
```

Comment:
🙈I can't really deal with unit tests on private methods. are we able to test this in a more integrated way from some public interface at any stack level? I mean, can we try to change the status when that `if` is suppose to return `false` and expect it not to trigger the change of status and vice-versa?

Reply:
We have very similar tests for the new registration workflow, which is where this came from: https://github.com/DEFRA/waste-exemptions-engine/blob/master/spec/models/waste_exemptions_engine/new_registration_workflow_state/location_form_spec.rb

I think the expectation on line 27 is the public interface test you're after. I'm inclined to agree with you though. At first I thought it wasn't hurting anything but it's not really adding much either. So I'll rip this out now and get rid of the expectations from the other file in a separate PR.

Reply:
Gone: https://github.com/DEFRA/waste-exemptions-engine/pull/139/commits/8bc4dac5f9ee6425024001dd18e10e950074929a

Reply:
And the related PR: https://github.com/DEFRA/waste-exemptions-engine/pull/146